kumascript/src/errors.js

```diff
@@ -16,13 +16,46 @@ class SourceCodeError {
     // instances of this class. Otherwise we'd need to monkey-patch
     // the `.toJSON` of `Error` which feels fragile.
     this.errorMessage = error.message;
+    this.offset = 0;
     this.line = line;
     this.column = column;
     this.macroName = macroName;
     this.sourceContext = this.getSourceContext(source);
     this.fatal = fatal;
   }
 
+  updateOffset(value) {
+    // Update the "offset" property to account for things like front-matter in the
+    // source. If the offset changes, this method will update related information.
+    // NOTE: We're not using a getter/setter for "offset", which would be a more
+    //       robust interface, so that the code that converts this instance to/from
+    //       JSON can remain simple.
+    const cleanValue = parseInt(value);
+    if (cleanValue >= 0 && this.offset !== cleanValue) {
```

Comment:
nit: wondering if `parseInt` and the `>= 0` check are too defensive. Given that `frontMatterOffset` should always be a number (or at the very least turned into one by `- 1`) and every doc having a frontmatter, thus the offset being at least ~4. But yeah, I'm nitting.